src/main.rs

```diff
@@ -28,19 +37,45 @@ struct CompilerOptions {
     /// Set the maximum stack size for the current running instance.
     #[clap(short, long, default_value = "10000")]
     stack_size: u32,
+
+    /// Run the compiler in debug mode
+    #[clap(short, long)]
+    debug: bool,
 }
 
 fn main() {
     let opts: CompilerOptions = CompilerOptions::parse();
 
-    println!("Stack_size is {}", opts.stack_size);
+    // print recieved cmdargs, if debug is specified
+    if opts.debug {
+        println!("Stack_size is {}", opts.stack_size);
 
-    for path in opts.includes.into_iter() {
-        println!("Running with {}", path);
+        for path in opts.includes.into_iter() {
+            println!("Running with {}", path);
+        }
     }
 
     match opts.execute {
-        Some(path) => println!("Are we executing -> {}", path),
+        Some(path) => match fs::canonicalize(&path) {
+            Ok(c) => {
+                // Resolve the module path
+                let contents = fs::read_to_string(c.to_str().expect("failed to convert"))
+                    .unwrap_or_else(|e| {
+                        report_error(
+                            ErrorType::IoError,
+                            format!("Couldn't read file '{}'\n{}", path, e),
+                        );
+                        exit(-1);
+                    });
+
+                // parse the given module
+                let _ = HashParser::parse(Rule::module, &contents).unwrap_or_else(|e| {
```

Comment:
We will probably wanna change this to use our AST after the "translation" is done.

Reply:
Indeed, this was primarily set up for testing purposes.